src/Init/Control/Basic.lean

```diff
@@ -146,12 +158,33 @@ in a new nested metavariable context. We can lift this to `withNewMctxDepth : n
 Which means that we can also run `withNewMctxDepth` in the `Tactic` monad without needing to
 faff around with lifts and all the other boilerplate needed in `mapped_foo`.
 
+## Relationship to `MonadFunctor`
+
+A stricter form of `MonadControl` is `MonadFunctor`, which defines
+`monadMap {α} : (∀ {β}, m β → m β) → n α → n α`. Using `monadMap` it is also possible to define `mapped_foo` above.
+However there are some mappings which can't be derived using `MonadFunctor`. For example:
+
+```lean,ignore
+ @[inline] def map1MetaM [MonadControlT MetaM m] [Monad m] (f : forall {α}, (β → MetaM α) → MetaM α) {α} (k : β → m α) : m α :=
+   control fun runInBase => f fun b => runInBase <| k b
+
+ @[inline] def map2MetaM [MonadControlT MetaM m] [Monad m] (f : forall {α}, (β → γ → MetaM α) → MetaM α) {α} (k : β → γ → m α) : m α :=
+   control fun runInBase => f fun b c => runInBase <| k b c
+```
+
+In these examples, `MonadControl` is needed because the lifted function
+needs to be all-quantified over the monadic return value,
+as that is where the surrounding monad's (`n`) state is stored.
+In the Lean monad-transformer stacks, there are no `MonadFunctor`s that
+are not also `MonadControl`s and so `MonadFunctor` is not used.
```

Comment:
Oh, `monadMap` *is* used in plenty places! It could be argued whether we still need `MonadFunctor`, but let's keep that discussion out of this docstring.